### PR TITLE
redpanda: migrate mount helpers

### PR DESCRIPTION
**Pull request by chrisseto**
#### 910ef39ce438f42fd19545ce9e7c5ad88f373706 redpanda: migrate mount helpers

This commit migrates `common-volumes`, `default-volumes`,
`default-mounts`, and `common-mounts` to go.

### CODE REVIEW

File: charts/redpanda/helpers.go [hunk 1]
```diff
@@ -172,6 +172,129 @@ func ClientAuthRequired(dot *helmette.Dot) bool {
...
+
+		for _, name := range certNames {
+			cert := values.TLS.Certs[name]
+
+			volumes = append(volumes, corev1.Volume{
+				Name: fmt.Sprintf("redpanda-%s-cert", name),
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: CertSecretName(dot, name, &cert),
+						// DefaultMode: ptr.To[int32](0o440),
```
Comment (RafalKorepta) [inline charts/redpanda/helpers.go:255]: Could this be removed if there is DefaultMode that repeats `0o440` access mode?

File: charts/redpanda/helpers.go [hunk 2]
```diff
@@ -172,6 +172,129 @@ func ClientAuthRequired(dot *helmette.Dot) bool {
...
+			})
+		}
+
+		if ClientAuthRequired(dot) {
+			volumes = append(volumes, corev1.Volume{
+				Name: "mtls-client",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: fmt.Sprintf("%s-client", Fullname(dot)),
+						// DefaultMode: ptr.To[int32](0o440),
```
Comment (RafalKorepta) [inline charts/redpanda/helpers.go:268]: ditto

**Review by RafalKorepta — APPROVED** [review]
LGTM. I left only nits

File: charts/redpanda/helpers.go (hunk 1 above)
Comment (chrisseto) [reply charts/redpanda/helpers.go]: Whoops! Removed.

**Pull request merged**